### PR TITLE
mds: Include more information in print statement in MDSRank.cc

### PR DESCRIPTION
**Pull request by Patty8122**
mds/MDSRank.cc: Updated the print statement to include filepath(root.c_str()).
From ss << "root inode is not in cache"; to ss << "inode for path '" << filepath(root.c_str()) << "' is not in cache as suggested in the issue description
Fixes: https://tracker.ceph.com/issues/57059
Signed-off-by: Patty8122 <divyapattisapu@uchicago.edu>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

Just a console logging update.
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
Thanks @Patty8122 ! We'll take a look.

File: src/mds/MDSRank.cc [hunk 1]
```diff
@@ -3183,7 +3183,7 @@ void MDSRank::command_dump_tree(const cmdmap_t &cmdmap, std::ostream &ss, Format
   std::lock_guard l(mds_lock);
   CInode *in = mdcache->cache_traverse(filepath(root.c_str()));
   if (!in) {
-    ss << "root inode is not in cache";
+    ss << "inode for path '" << filepath(root.c_str()) << "' is not in cache";
```
Comment (dparmar18) [inline src/mds/MDSRank.cc:3186]: when we say we are traversing through mdcache, the "root inode" is self explanatory that it is `/`.

Comment (ljflores) [reply src/mds/MDSRank.cc:3186]: Hey @dparmar18, see the original description for https://tracker.ceph.com/issues/57059. Perhaps there is still a way to make this message more informative?

Comment (lxbsz) [reply src/mds/MDSRank.cc:3186]: > when we say we are traversing through mdcache, the "root inode" is self explanatory that it is `/`.

IMO the `root` here means the root of a path passed from the `dump tree` command, such as a path `a/b/c/d/`, the `root` here should be `a/`.

Comment (dparmar18) [reply src/mds/MDSRank.cc:3186]: hmm so this is quite misleading, say for e.g. `ceph tell mds.0 dump tree foo` tells `root inode is not in cache` which definitely is misleading and needs to be fixed. My bad i didn't take a look at the tracker, thanks @ljflores @lxbsz

**Comment by dparmar18** [comment]
the first commit b30c35019d0128a0cf840b43d3059392effbb52c looks unrelated though

**Comment by ljflores** [comment]
@Patty8122 I fixed your branch; you had an extra unrelated commit.

Here are the steps I took:

```
git log
git rebase -i 9fedc1e062027dbce66747e5d0dc11319615ab8a
Wrote "drop" next to commit b30c35019d0128a0cf840b43d3059392effbb52c instead of "pick"
Saved the file (`:wq`)
git push -f Patty8122 adding-useful-print
```

**Review by gregsfortytwo — APPROVED** [review]
LGTM

**Review by dparmar18 — APPROVED** [review]
It was a long weekend, apologies for the delay. looks good now post removal of the unrelated commit!

**Comment by ljflores** [comment]
@Patty8122 everything is good on your end now. Now, this PR needs to go through additional integration testing. If you're curious, you can read about it here: https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro/

The workflow is that your PR will be tagged with a label that says some variation of "wip-yuri", and @yuriw (the QA lead) will run it through integration testing. If there is any kind of problem, someone will comment on the PR and tell you if something additional needs to be fixed. But if everything passes, someone will comment that the PR is approved.

Then it will be merged! Let me know if you have questions about that process.

**Review by rishabh-d-dave — APPROVED** [review]

**Review by anthonyeleven — APPROVED** [review]

**Review by rishabh-d-dave — APPROVED** [review]
CephFS integration tests ran successfully - https://tracker.ceph.com/projects/cephfs/wiki/Main#3-Nov-2023

**Pull request merged**